### PR TITLE
Add regression tests for [TG-8284]

### PR DESCRIPTION
**Pull request by yumibagge**
This PR addresses
- regression folder rename
- add regression test for models-library
- add regression test for Character

- n/a Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5ef594d5d7f50471e66d0533c882a12a4de303c0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/120229016

**Review by danpoe — APPROVED** [review]

File: jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+public class Main {
+
+  public void constantIndexOf() {
+    String s1 = "abcabc";
+    Character b = 'b';
+    assert s1.indexOf(b) == 1;
+    assert s1.indexOf(b, -10) == 1;
+    assert s1.indexOf(b, 3) == 4;
+    assert s1.indexOf(b, 10) == -1;
```
Comment (allredj) [inline jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java]: In cbmc, it's generally not enough to check for assertion success. After all, cbmc's job is to violate assertions, so if there's nothing to violate, we can't tell if cbmc did anything at all.
I recommend looking at the tests in https://github.com/diffblue/cbmc/tree/develop/jbmc/regression/jbmc-strings/CompareToConstantEvaluation that exercise failing assertions.

**Review by allredj — APPROVED** [review]

File: jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java (hunk 1 above)
Comment (yumibagge) [reply jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java]: Thanks for the great advice and reference. Indeed, I had a test.desc for this test that does not use models-library, which is verification is expected to be failed. Is it better to add the test? The detailed verification failure is https://diffblue.atlassian.net/browse/TG-8284?focusedCommentId=23418&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-23418

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 723312ff2387b3c159886f69d21f5dca0f6028ab).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/120479078

File: jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java (hunk 1 above)
Comment (allredj) [reply jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java]: That test you link doesn't look like one you would want to add. It fails on a NullPointerException, which is not what you're looking for. Happy to discuss in person.

Comment (yumibagge) [reply jbmc/regression/jbmc-strings/IndexOfConstantEvaluation05-Charactor/Main.java]: @allredj , I added a test to check assertion failure. Could you re-review?

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 9cddf5dddfdc3d424545ff5751cd90b64f06a053).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/120755220

File: jbmc/regression/jbmc-strings/IndexOfConstantEvaluation06-AssertionFailure/test.desc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CORE symex-driven-lazy-loading-expected-failure
+Main.class
+--function Main.constantIndexOf
+^Generated [0-9]+ VCC\(s\), 1 remaining after simplification$
+^VERIFICATION FAILED$
```
Comment (allredj) [inline jbmc/regression/jbmc-strings/IndexOfConstantEvaluation06-AssertionFailure/test.desc:5]: I think you should specify which assertion you want to fail. It's probably good to use the `--property` option too, as you don't want to analyse the assertions that you expect to be successful.
Look at the tests in https://github.com/diffblue/cbmc/tree/develop/jbmc/regression/jbmc-strings/CompareToConstantEvaluation

Comment (yumibagge) [reply jbmc/regression/jbmc-strings/IndexOfConstantEvaluation06-AssertionFailure/test.desc:5]: Yes ... it was pointed out by @danpoe when he explained to me about regression tests. I was just being lazy :sweat_smile: property added.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 535299620234f9dd6259437a41e4a1d02e72344c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/120836927

**Pull request merged**